### ISSUE TITLE
[BUG]: Getting inconsistent results from anti_spoofing for real images

### ISSUE THREAD

**Issue by konstantin-nachev** [serengil/deepface#1399]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [X] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

v0.0.93

### Python version

3.12.6

### Operating System

macOS Sequoia Version 15.2 (24C101)

### Dependencies

absl-py==2.1.0
apispec==6.8.0
astunparse==1.6.3
attrs==24.2.0
beautifulsoup4==4.12.3
blinker==1.9.0
botocore==1.35.77
cachetools==5.5.0
certifi==2024.8.30
cffi==1.17.1
charset-normalizer==3.4.0
click==8.1.7
contourpy==1.3.1
cycler==0.12.1
deepface==0.0.93
dlib==19.24.6
facenet-pytorch==2.6.0
filelock==3.16.1
fire==0.7.0
Flask==3.1.0
Flask-Cors==5.0.0
flask-smorest==0.45.0
flatbuffers==24.3.25
fonttools==4.55.3
fsspec==2024.10.0
gast==0.6.0
gdown==5.2.0
google-api-core==2.24.0
google-auth==2.36.0
google-cloud-core==2.4.1
google-cloud-firestore==2.19.0
google-pasta==0.2.0
googleapis-common-protos==1.66.0
grpcio==1.68.1
grpcio-status==1.68.1
gunicorn==23.0.0
h5py==3.12.1
idna==3.10
itsdangerous==2.2.0
jax==0.4.37
jaxlib==0.4.36
Jinja2==3.1.4
jmespath==1.0.1
joblib==1.4.2
keras==3.7.0
kiwisolver==1.4.7
libclang==18.1.1
lz4==4.3.3
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
marshmallow==3.23.1
matplotlib==3.9.3
mdurl==0.1.2
mediapipe==0.10.18
ml-dtypes==0.4.1
mpmath==1.3.0
mtcnn==1.0.0
namex==0.0.8
networkx==3.4.2
numpy==1.26.4
opencv-contrib-python==4.10.0.84
opencv-python==4.10.0.84
opt_einsum==3.4.0
optree==0.13.1
packaging==24.2
pandas==2.2.3
pillow==10.2.0
proto-plus==1.25.0
protobuf==4.25.5
pyasn1==0.6.1
pyasn1_modules==0.4.1
pycparser==2.22
Pygments==2.18.0
pyparsing==3.2.0
PySocks==1.7.1
python-dateutil==2.9.0.post0
pytz==2024.2
requests==2.32.3
retina-face==0.0.17
rich==13.9.4
rsa==4.9
scipy==1.14.1
sentencepiece==0.2.0
setuptools==75.6.0
six==1.17.0
sounddevice==0.5.1
soupsieve==2.6
sympy==1.13.1
tensorboard==2.18.0
tensorboard-data-server==0.7.2
tensorflow==2.18.0
termcolor==2.5.0
tf_keras==2.18.0
torch==2.2.2
torchvision==0.17.2
tqdm==4.67.1
typing_extensions==4.12.2
tzdata==2024.2
urllib3==2.2.3
webargs==8.6.0
Werkzeug==3.1.3
wheel==0.45.1
wrapt==1.17.0

### Reproducible example

```Python
result = DeepFace.extract_faces(
            img_path=f"data:image/jpeg;base64,{base64_image}",
            detector_backend="mtcnn",
            anti_spoofing=True,
            enforce_detection=True
        )
        
        # Print results
        print("Results:")
        print("-" * 50)
        for idx, face in enumerate(result, 1):
            print(f"\nFace #{idx}:")
            print(f"Facebox {bounding_box}")
            print(f"Is Real: {face.get('is_real', 'Unknown')}")
            print(f"Anti-spoofing Score: {face.get('antispoof_score', 'Unknown')}")
            
            facial_area = face.get('facial_area', {})
            if facial_area:
                print(f"Face Location: x={facial_area.get('x')}, y={facial_area.get('y')}, "
                      f"w={facial_area.get('w')}, h={facial_area.get('h')}")
```

### Relevant Log Output

Results:
--------------------------------------------------

Face #1:
Is Real: False
Anti-spoofing Score: 0.9958124756813049
Face Location: x=188, y=100, w=214, h=289

### Expected Result

When using the deepface antispoofing option we should get consistent results for basic images of real people.

### What happened instead?

Most of the time it returns a person is not real, even though this is not the case.

### Additional Info

In my project, we are using DeepFace to detect faces and possible spoofing in images collected from the front camera of an Android Tablet. For the same camera device and the same person we are getting a lot of inconsistent results for the spoofing detection - it reports as spoof real people. I am attaching one of the images with which I am testing. 
![asen_shows_always_spoof](https://github.com/user-attachments/assets/55c53110-8a59-49cc-bd6c-254a1c27aeb5)

**Comment by serengil** [comment]
individual misclassifications don't make a model wrong. this is the nature of machine learning models.

**Comment by brnyza** [comment]
I believe the reflection on the image is causing the model to mistake it for a static fake image from a mobile phone. I'm having the same issue with users who wear glasses.

<img width="329" height="510" alt="Image" src="https://github.com/user-attachments/assets/3806f577-2d18-41b6-9873-f3b4ad383825" />
